src/blocks/time.rs

```diff
@@ -78,44 +97,59 @@ impl ConfigBlock for Time {
     ) -> Result<Self> {
         let i = Uuid::new_v4().to_simple().to_string();
         Ok(Time {
+            time: Local::now(),
             id: i.clone(),
             format: block_config.format,
-            time: ButtonWidget::new(config, i.as_str())
+            format_short: block_config.format_short,
+            widget: ButtonWidget::new(config, i.as_str())
                 .with_text("")
+                .with_short_text("")
                 .with_icon("time"),
             update_interval: block_config.interval,
             on_click: block_config.on_click,
             timezone: block_config.timezone,
+            width: block_config.width,
         })
     }
 }
 
 impl Block for Time {
     fn update(&mut self) -> Result<Option<Update>> {
-        let time = match self.timezone {
-            Some(tz) => Utc::now().with_timezone(&tz).format(&self.format),
-            None => Local::now().format(&self.format),
+        self.time = match self.timezone {
+            Some(tz) => Local.from_utc_datetime(&Utc::now().with_timezone(&tz).naive_local()),
+            None => Local::now(),
         };
-        self.time.set_text(format!("{}", time));
+        self.widget.set_text_with_width(
+            &self.width,
+            format!("{}", self.time.format(&self.format)),
+            format!("{}", self.time.format(&self.format_short)),
+        );
         Ok(Some(self.update_interval.into()))
     }
 
     fn click(&mut self, e: &I3BarEvent) -> Result<()> {
         if let Some(ref name) = e.name {
             if name.as_str() == self.id {
-                if let MouseButton::Left = e.button {
-                    if let Some(ref cmd) = self.on_click {
-                        spawn_child_async("sh", &["-c", cmd])
-                            .block_error("time", "could not spawn child")?;
+                match e.button {
+                    MouseButton::Left => {
+                        if let Some(ref cmd) = self.on_click {
+                            spawn_child_async("sh", &["-c", cmd])
+                                .block_error("time", "could not spawn child")?;
+                        }
+                    }
+                    MouseButton::Right | MouseButton::Middle => {
+                        self.width = self.width.next();
+                        let _ = self.update();
```

Comment:
@ammgws Is it better to do the update via a `Sender` like it is done in custom block? 
```
            self.tx_update_request.send(Task {
                id: self.id.clone(),
                update_time: Instant::now(),
            })?;
```